### ISSUE TITLE
[4.9.4] CWO_CLOSED_WITHOUT_OPENED false positive

### ISSUE THREAD

**Issue by ben-manes** [spotbugs/spotbugs#3616]
Upgrading from 4.9.3 and I receive this when the usages match the required pattern,

<img width="1136" height="299" alt="Image" src="https://github.com/user-attachments/assets/ed21a4a2-1bfc-4573-89b2-ac3a4c4ff5bf" />

For example, [performCleanUp()](https://github.com/ben-manes/caffeine/blob/38dd616b945749edcc102642adb1b08274ae3a88/caffeine/src/main/java/com/github/benmanes/caffeine/cache/BoundedLocalCache.java#L1656-L1669) is implemented as,

```java
/**
  * Performs the maintenance work, blocking until the lock is acquired.
  *
  * @param task an additional pending task to run, or {@code null} if not present
  */
void performCleanUp(@Nullable Runnable task) {
  evictionLock.lock();
  try {
    maintenance(task);
  } finally {
    evictionLock.unlock();
  }
  rescheduleCleanUpIfIncomplete();
}
```

And that should be recognized as similar. At first I thought it was confused by the trailing code, but it flags the unit test [evict_alreadyRemoved](https://github.com/ben-manes/caffeine/blob/38dd616b945749edcc102642adb1b08274ae3a88/caffeine/src/test/java/com/github/benmanes/caffeine/cache/BoundedLocalCacheTest.java#L891-L931) which had none. Another possibility was if it required the lock's final field be held in a local variable, which was a j.u.c. quick due to JIT inlining quirks in JDK5, but that too failed. The rule's [pull request](https://github.com/spotbugs/spotbugs/pull/3385) unit test includes a catch clause in its successful case, but adding that likewise did nothing to help pass in my runs.

<img width="1721" height="332" alt="Image" src="https://github.com/user-attachments/assets/b39036e0-4824-46b6-b371-37ddf3ecffc8" />

**Comment by ben-manes** [comment]
Thanks for the fix! In v4.9.5 this occurred in one instance.

<img width="833" height="169" alt="Image" src="https://github.com/user-attachments/assets/8a923f0d-627f-408e-b42d-80dbfb6b7222" />

```java
/** Clears the internal state of the cache and becomes unusable. */
public static void destroy(Cache<?, ?> cache) {
  var local = (cache instanceof AbstractCacheView<?, ?>)
      ? ((AbstractCacheView<?, ?>) cache).asyncCache().cache()
      : (LocalCache<?, ?>) cache.asMap();
  if (local instanceof UnboundedLocalCache) {
    var unbounded = (UnboundedLocalCache<?, ?>) local;
    unbounded.data.clear();
  } else if (local instanceof BoundedLocalCache) {
    @SuppressWarnings("unchecked")
    var bounded = (BoundedLocalCache<Object, Object>) local;
    bounded.evictionLock.lock();
    try {
      for (var node : bounded.data.values()) {
        destroyNode(bounded, node);
      }
      if (bounded.expiresVariable()) {
        destroyTimerWheel(bounded);
      }
      bounded.data.clear();
      bounded.writeBuffer.clear();
      bounded.readBuffer.drainTo(e -> {});
    } finally {
      bounded.evictionLock.unlock();
    }
  }
}
```
